### DISCUSSION TITLE
Apply-PnPProvisioningTemplate

### DISCUSSION THREAD

**Discussion by msiejda** [pnp/powershell#254] [Q&A] [answered]
Hello,

Are You planning to add Apply-PnPProvisioningTemplate function to PnP.PowerShell?

I was using SharePointPnPPowerShellOnline with Powershell 5.1 and I managed to built some scripts. I wanted to improve them with Azure functions but there is an issue that you can use only Powershell 7.0 with Azure function apps.
There are errors like "The 'Apply-PnPProvisioningTemplate' command was found in the module 'SharePointPnPPowerShellOnline', but the module could not be loaded."
I've also tried to reproduce issue locally on Powershell 7.0 where is different error "The remote server returned an error: (400) Bad Request."
Same script using Apply-SPOProvisioningTemplate works perfect on Powershell 5.1

Or maybe You can help me out with some workaround?

Best regards,
Michal Siejda

**Comment by erwinvanhunen** [comment, marked as answer]
It's there, but we renamed it. It's called `Invoke-PnPSiteTemplate` now. Same parameters, same functionality.

**Comment by msiejda** [comment]
Thank You,

it works now. I would like to contribute and share the code that works now.

I was folowing this manual [https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/site-design-pnp-provisioning](url) and then faced problem described above. I have been thinking that Invoke-PnPSiteTemplate is only providing new templates but like Ervin wrote it work same as Apply-PnPProvisioningTemplate.

One more thing - I had to convert my template using Convert-PnPProvisioningTemplate - here's manual [https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/convert-pnpprovisioningtemplate?view=sharepoint-ps](url)

The other thing I had to reslove and had problems was how to define $env: parameters - You do it in function app / configuration / app settings

```
using namespace System.Net

# Input bindings are passed in via param block.
param($QueueItem, $TriggerMetadata)
$Request = $QueueItem
Write-Output $("QueueItem: {0}" -f $QueueItem);

# Write to the Azure Functions log stream.
Write-Host "PowerShell HTTP trigger function processed a request."

# Interact with query parameters or the body of the request.
$tenant = $QueueItem.tenant
Write-Output $("Tenant: {0}" -f $tenant);
$targetSite = "/sites/{0}" -f $QueueItem.GroupName;
$GroupName = $QueueItem.GroupName
Write-Output $("GroupName: {0}" -f $GroupName);
$webUrl = "https://{0}.sharepoint.com{1}/" -f $tenant, $targetSite;
$PlanID = $QueueItem.PlanID
Write-Output $("PlanID: {0}" -f $PlanID);

Write-Output $("Connecting to {0}" -f $webUrl);
Connect-PnPOnline -url $webUrl -Credentials (New-Object System.Management.Automation.PSCredential ($env:SPO_U, (ConvertTo-SecureString $env:SPO_P -AsPlainText -Force)))
$web = Get-PnPWeb
Write-Output $("Current web {0}" -f $web);
$templateFile = "D:\home\site\wwwroot\InvokePnPSiteTemplate\newtemplate.xml"
Write-Output $("Loaded script {0}" -f $templateFile);

Invoke-PnPSiteTemplate -Path $templateFile -Parameters @{"O365TenantName"="$tenant";"GroupName"="$GroupName";"PlanID"="$PlanID"}
```
